ledger/internal/eval.go

```diff
@@ -1483,6 +1483,7 @@ type evalTxValidator struct {
 	txcache          verify.VerifiedTransactionCache
 	block            bookkeeping.Block
 	verificationPool execpool.BacklogPool
+	ledger           logic.LedgerForLogic
```

Comment:
I don't see where this field is ever assigned

Reply:
Agreed. What on earth.

Reply:
It seems as though LogicSigs can be evaluated along two paths, and I've clearly not fixed the path that uses `evalTxValidator`. Before I fix it, I'd like to find or write failing tests.

Reply:
@algorandskiy do we have tests evalTxValidator? It looks to me like it's use in txn validation, so I don't understand how my code is passing most tests, including e2e tests, without a way to get block headers.

Reply:
I suppose the only ones that would fail are ones with a `block` of `txn FirstValidTime` in them (since only they access SigLedger).  But we have an e2e test that does that, and it passes.

Reply:
I'm committing a check in `EvalSignature` for SigLedger == nil. I suppose we should have it anyway, but my goal is to see if maybe we do have a test that exercises this code, but it was not failing because it was not using the new opcodes, so it didn't complain about the nil SigLedger.